packages/gafl-webapp-service/src/processors/payment.js

```diff
@@ -50,17 +50,21 @@ export const preparePayment = (request, transaction) => {
     }
   }
 
+  if (recurring) {
+    result.set_up_agreement = transaction.agreementId
+  }
```

Comment:
Could we modify the logic a bit - so that if agreementId is present, it is included, rather than relying on the flag variable? Seems like it might be a bit of a 'gotcha' further down the line.